src-tauri/src/note/note_api.rs

```diff
@@ -0,0 +1,45 @@
+use crate::api::HttpClient;
+use anyhow::Result;
+
+use super::Note;
+
+/// Handler to communicate with the server through the notes API.
+#[derive(Debug)]
+pub struct NoteApi {
+    client: HttpClient,
+    base_url: String,
+}
+
+impl Default for NoteApi {
+    fn default() -> Self {
+        Self {
+            client: HttpClient::default(),
+            base_url: "http://localhost:3000/notes".to_string(),
```

Comment:
Move root to .env

Have the client be an already created instance rather than creating it in every file

Have the root inside the instance and give /notes independently

Reply:
> Move root to .env
> 
> Have the client be an already created instance rather than creating it in every file
> 
> Have the root inside the instance and give /notes independently

The configuration aspect will be treated within its own issue. This will include the root url.
For now the HttpClient will be created from the manager.